### PR TITLE
Define lc0 interoperability and chess-decision scope

### LINKED ISSUE #129: [Design] Define lc0 interoperability and chess-decision scope
## Goal

Define one coherent product boundary with two mutually supporting responsibilities:

1. make lc0-family models portable and operable in the PyTorch ecosystem; and
2. express evaluator and search behavior in chess-domain terms.

The library must remain compatible with arbitrary hooks, patches, probes, attribution tools, and other external interpretability workflows without owning those methods.

## Scope

- Write the canonical one-sentence mission and supported use cases.
- Define the model boundary: chess positions in; standardized policy/WDL/value/MLH outputs out.
- Identify the lc0 interoperability substrate that remains first-class: encoding, policy vocabulary, model conversion/loading, TensorDict modules, batching, legal masking, and conformance.
- Define chess-decision analysis responsibilities: position facts, move/variation evidence, counterfactuals, and search traces/comparisons.
- State explicit non-goals: chess rules, production engine search, generic hooks, attribution, probing, SAEs/transcoders, and natural-language coaching.
- Produce a disposition table for every current public module and symbol: retain, refocus, deprecate, or internalize.
- Define the public API and compatibility policy before adding new feature surface.

## Acceptance criteria

- [ ] README and contributor documentation use one consistent scope statement.
- [ ] An architecture map shows the boundary between `python-chess`, lc0/model adapters, external interpretability tools, search, and chess-semantic analysis.
- [ ] Every existing public module has an explicit disposition and rationale.
- [ ] External interpretability frameworks appear as integrations/examples, not core abstractions.
- [ ] No implementation-specific neural-interpretability method is required by the core API.
- [ ] The next backlog stages can be evaluated against the documented ownership boundary.

## Assessment checkpoint A — positioning

Before #130 begins, verify that:

- a user can explain why model conversion/TensorDict support and chess/search analysis belong in the same package;
- an arbitrarily instrumented model can preserve the evaluator contract without changes to `lczerolens`; and
- no retained module lacks a clear role in the mission.

If any answer is unclear, revise the scope before expanding the API.

Depends on #128.

**Roadmap order:** 1 — positioning checkpoint.

### PR DESCRIPTION
**Pull request by Xmaster6y**
## Summary

- define the canonical lc0 interoperability and chess-decision analysis mission
- add the evaluator contract, architecture boundary, non-goals, public-surface disposition, and compatibility policy
- align README, docs entry points, and contributor guidance with the boundary

## Why

The package previously described itself as a home for framework-agnostic interpretability. This makes ownership explicit: lczerolens owns lc0 model interoperability and chess-domain decision evidence, while external tools own hooks, probes, patches, and attribution methods.

## Validation

- `git diff --check`
- `uv run --group docs sphinx-build -b html docs/source /private/tmp/lczerolens-issue-129-docs-final` (succeeds; existing project warnings remain)

Closes #129

### CODE REVIEW

**Review by Xmaster6y — COMMENTED** [review]
The positioning is coherent and substantially satisfies #129: the two pillars reinforce each other, generic neural-interpretability methods are correctly externalized, and the public-surface disposition gives later issues a useful constraint.

I found three documentation-contract inconsistencies to resolve before treating Checkpoint A as passed; I left them inline. One additional consistency fix is needed in `README.md`: the “best move” example still calls `argmax()` over all 1858 policy logits, although this document correctly says consumers must mask to legal indices. Please update that example to gather `board.get_legal_indices()` and decode the winning legal index.

After those changes, I think this is the right foundation for #130.

File: docs/source/scope.rst [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+Supported use cases
+-------------------
+
+lczerolens supports:
+
+* loading or converting lc0-family weights and evaluating one position or a
+  batch in PyTorch;
+* encoding positions and mapping policy indices to legal chess moves;
+* consuming a standardized policy, WDL, value, and MLH evaluator result;
+* recording and comparing position facts, moves, variations, counterfactual
```
Comment (Xmaster6y) [inline docs/source/scope.rst]: This presents roadmap work as functionality available today, but there is currently no counterfactual API or search-trace schema; those are #131, #133, #140, and #141. Because this is user-facing compatibility documentation, please separate **current capabilities** from **planned in-scope capabilities** (or say “the planned public surface includes …”) so v0.4 users are not promised APIs that do not exist yet.

File: docs/source/scope.rst [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+  positions, and search traces; and
+* passing an evaluator through arbitrary external instrumentation while
+  retaining the same input/output contract.
+
+Evaluator contract
+------------------
+
+The model boundary is ``LczeroBoard`` positions in and a ``TensorDict`` out.
```
Comment (Xmaster6y) [inline docs/source/scope.rst]: This is narrower than the current callable contract: `LczeroModel.forward()` accepts one `LczeroBoard`, an iterable of boards, a 3D/4D tensor, or a `TensorDict`. It also returns raw 1858-way policy values; the model wrapper does not perform legal masking. Since this section is establishing a compatibility contract, please distinguish the conceptual chess boundary from the accepted runtime inputs and state that legal masking is currently a downstream operation (until #130 deliberately changes that contract).

File: docs/source/scope.rst [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+Architecture boundary
+---------------------
+
+.. code-block:: text
+
+   python-chess                      external interpretability tools
+   rules, FEN, legal moves           hooks, patches, probes, attribution
+          |                                          |
+          v                                          | instruments only
+   LczeroBoard -- encoding / move vocabulary --------+
```
Comment (Xmaster6y) [inline docs/source/scope.rst]: The diagram currently joins “external interpretability tools” to `LczeroBoard`, implying that hooks/patches instrument the board encoding layer. The agreed boundary is that these tools wrap or instrument the PyTorch evaluator/model while preserving its I/O contract. Please move this arrow to the lc0 adapter/evaluator portion of the diagram; that distinction is central to the positioning.
